src/EditorFeatures/Core.Wpf/NavigateTo/NavigateToItemProvider.cs

```diff
@@ -30,21 +32,60 @@ internal partial class NavigateToItemProvider : INavigateToItemProvider2
             _displayFactory = new NavigateToItemDisplayFactory();
         }
 
-        public ISet<string> KindsProvided { get; } = ImmutableHashSet.Create(
-            NavigateToItemKind.Class,
-            NavigateToItemKind.Constant,
-            NavigateToItemKind.Delegate,
-            NavigateToItemKind.Enum,
-            NavigateToItemKind.EnumItem,
-            NavigateToItemKind.Event,
-            NavigateToItemKind.Field,
-            NavigateToItemKind.Interface,
-            NavigateToItemKind.Method,
-            NavigateToItemKind.Module,
-            NavigateToItemKind.Property,
-            NavigateToItemKind.Structure);
-
-        public bool CanFilter => true;
+        ISet<string> INavigateToItemProvider2.KindsProvided => KindsProvided;
+
+        public ImmutableHashSet<string> KindsProvided
+        {
+            get
+            {
+                var result = ImmutableHashSet.Create<string>(StringComparer.Ordinal);
+                foreach (var project in _workspace.CurrentSolution.Projects)
+                {
+                    var navigateToSearchService = project.LanguageServices.GetService<INavigateToSearchService_RemoveInterfaceAboveAndRenameThisAfterInternalsVisibleToUsersUpdate>();
+                    if (navigateToSearchService != null)
+                    {
+                        result = result.Union(navigateToSearchService.KindsProvided);
```

Comment:
Use a builder?

Reply:
This is _supposed_ to be a NOP on each loop. It's not clear whether `ImmutableHashSet<T>` implements the necessary edge cases as fast paths.

1. When `result` is empty, the right hand side is returned
2. When the right hand side is empty, `result` is returned unaltered
3. In all other cases where the solution does not contain both C#/VB and F#/TypeScript projects, `result` and the right hand side reference the same object